x/ibc/03-connection/keeper/verify.go

```diff
@@ -19,9 +19,10 @@ func (k Keeper) VerifyClientConsensusState(
 	proof commitment.ProofI,
 	consensusState clientexported.ConsensusState,
 ) error {
-	clientState, found := k.clientKeeper.GetClientState(ctx, connection.GetClientID())
+	clientID := connection.GetCounterparty().GetClientID()
```

Comment:
This line seems wrong to me - we should never be fetching the local client state by looking up the counterparty's client identifier. The only thing we need the counterparty's client identifier for is to check whether the correct client state is stored under it (for this chain's consensus algorithm).